File: src/test/java/xtdb/api/XtdbHelloWorld.java
package xtdb.api;

import java.sql.DriverManager;
import java.sql.SQLException;

public class XtdbHelloWorld {
    public static void main(String[] args) throws SQLException {
        try(var connection =
                DriverManager.getConnection("jdbc:postgresql://localhost:5432/xtdb", "xtdb", "xtdb");
            var statement = connection.createStatement()) {

            statement.execute("INSERT INTO users RECORDS {_id: 'jms', name: 'James'}, {_id: 'joe', name: 'Joe'}");

            try (var resultSet = statement.executeQuery("SELECT * FROM users")) {
                System.out.println("Users:");

                while (resultSet.next()) {
                    System.out.printf("  * %s: %s%n", resultSet.getString("_id"), resultSet.getString("name"));
                }
            }
        } catch (SQLException e) {
            e.printStackTrace();
            throw e;
        }
    }
}

File: src/test/java/xtdb/api/XtdbHelloWorld.kt
package xtdb.api

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import xtdb.test.NodeResolver
import java.sql.DriverManager

@ExtendWith(NodeResolver::class)
class XtdbHelloWorld {
    data class User(val id: String, val name: String)

    @Test
    fun `hello world`(xtdb: Xtdb) {
        xtdb.connection.use { conn ->
            conn.createStatement().use { statement ->
                statement.execute("INSERT INTO users RECORDS {_id: 'jms', name: 'James'}, {_id: 'joe', name: 'Joe'}")

                statement.executeQuery("SELECT * FROM users").use { rs ->

                    val users = mutableListOf<User>()

                    while (rs.next()) {
                        users.add(User(rs.getString("_id"), rs.getString("name")))
                    }

                    assertEquals(listOf(User("joe", "Joe"), User("jms", "James")), users)
                }
            }
        }
    }
}
